### DISCUSSION TITLE
Pipewire Virtual Sources

### DISCUSSION THREAD

**Discussion by ravenbird88** [baresip/baresip#3158] [Ideas]
Hello.
Pipewire provides two types of audio sources: "Audio/Source" and "Audio/Source/Virtual".
In version 3.16 only "Audio/Source" is parsed by "pipewire.c" module (line 120):
`	if (!str_cmp(media_class, "Audio/Source") && str_isset(node_name)) {`
It would be great to allow both types of sources. It can be very useful for some applications.

**Comment by cspiel1** [comment]
Thanks for the hint! Your welcome to open a PR. Or I'll try to do this in the next ~14 days.
